services/manifest-repo-export-service/pkg/repository/transformer.go

```diff
@@ -491,6 +491,105 @@ func (c *DeleteEnvironmentLock) Transform(
 	return fmt.Sprintf("Deleted lock %q on environment %q", c.LockId, c.Environment), nil
 }
 
+type CreateEnvironmentApplicationLock struct {
+	Authentication `json:"-"`
+	Environment    string `json:"env"`
+	Application    string `json:"app"`
+	LockId         string `json:"lockId"`
+	Message        string `json:"message"`
+}
+
+func (c *CreateEnvironmentApplicationLock) GetDBEventType() db.EventType {
+	return db.EvtCreateEnvironmentApplicationLock
+}
+
+func (c *CreateEnvironmentApplicationLock) Transform(
+	ctx context.Context,
+	state *State,
+	t TransformerContext,
+	transaction *sql.Tx,
+) (string, error) {
+	fs := state.Filesystem
+	envDir := fs.Join("environments", c.Environment)
+	if _, err := fs.Stat(envDir); err != nil {
+		return "", fmt.Errorf("error accessing dir %q: %w", envDir, err)
+	}
+
+	appDir := fs.Join(envDir, "applications", c.Application)
+	if err := fs.MkdirAll(appDir, 0777); err != nil {
+		return "", err
+	}
+
+	lock, err := state.DBHandler.DBSelectAppLock(ctx, transaction, c.Environment, c.Application, c.LockId)
+
+	if err != nil {
+		return "", err
+	}
+
+	if lock == nil {
```

Comment:
why is this an error? If there's no lock there, we should create a lock.

Reply:
This should never happen. However, this would indicate that we are trying to create a new application lock that was not written to the database. (esl event was written but for some reason the application_locks table was not). I think this should generate an error, but maybe it shouldn't? 
Writing things into the manifest without them being present on the database (especially locks) essentially makes it a 'forgotten', as Kuberpult does not fetch data from the manifest as long as DB is enable (on already implemented features)

Reply:
ah, right, it should have been written by the cd-service at this point.
Then I agree, it should be an error.